pkg/cmd/template/cmd_overlays_test.go

```diff
@@ -238,101 +240,531 @@ overlayed: true
 	}
 }
 
-func TestMultipleDataValuesOneEmptyAndOneNonEmpty(t *testing.T) {
-	yamlTplData := []byte(`#@ load("@ytt:data", "data")
+func TestDocumentOverlayArrayItems(t *testing.T) {
+	t.Run("append with and without 'append' annotation", func(t *testing.T) {
```

Comment:
The difference between the two tests is that one is overlaying onto a Document and one is overlaying onto Data Values. We test these separately because the way we match the overlay is different, for Documents:
```
#@overlay/match by=overlay.all
---
```
For Data values (No 'match' annotation because multi data values are an implied overlay):
```
#@data/values
---
```

Agreed on the `t.Run`. It's removed.